classes/customfield/cmsfield_handler.php

```diff
@@ -51,13 +51,15 @@ public function get_configuration_url(): \moodle_url {
     }
 
     /**
-     * Context that should be used for data stored for the given record
+     * Context that should be used for data stored for the given record.
+     * Uses the activity context.
      *
      * @param int $instanceid id of the instance or 0 if the instance is being created
      * @return \context
      */
     public function get_instance_context(int $instanceid = 0): \context {
-        return $this->get_configuration_context();
+        $modinfo = get_coursemodule_from_instance('cms', $instanceid);
```

Comment:
What happens here if `$instanceid = 0` does it fail/explode? Or is the understanding that `$instanceid` can never be 0?

Reply:
It's never zero.

Reply:
I put in a extra check, just in case.